deepcell/image_generators/cropping.py

```diff
@@ -0,0 +1,576 @@
+                                 X.shape, y.shape))
+
+        if X.ndim != 4:
+            raise ValueError('Input data in `SemanticIterator` '
+                             'should have rank 4. You passed an array '
+                             'with shape', X.shape)
+
+        self.x = np.asarray(X, dtype=K.floatx())
+        self.y = np.asarray(y, dtype='int32')
+
+        self.channel_axis = 3 if data_format == 'channels_last' else 1
+        self.image_data_generator = image_data_generator
+        self.data_format = data_format
+        self.save_to_dir = save_to_dir
+        self.save_prefix = save_prefix
+        self.save_format = save_format
+        self.min_objects = min_objects
+
+        self.y_semantic_list = []  # optional semantic segmentation targets
+
+        # set output size based on cropping or not
+        if crop_size is not None:
+            output_size = crop_size
+
+        else:
+            output_size = self.x.shape[1:3] if self.channel_axis == 3 else self.x.shape[2:4]
+
+        self.output_size = output_size
+        # Create a list of all the semantic targets. We need to be able
+        # to have multiple semantic heads
+        # Add all the keys that contain y_semantic
+
+        # Add transformed masks
+        for transform in transforms:
+            transform_kwargs = transforms_kwargs.get(transform, dict())
+            y_transform = _transform_masks(y, transform,
+                                           data_format=data_format,
+                                           **transform_kwargs)
+            if y_transform.shape[self.channel_axis] > 1:
+                y_transform = np.asarray(y_transform, dtype='int32')
+            elif y_transform.shape[self.channel_axis] == 1:
+                y_transform = np.asarray(y_transform, dtype=K.floatx())
+            self.y_semantic_list.append(y_transform)
+
+        invalid_batches = []
+
+        # Remove images with small numbers of cells
+        for b in range(self.x.shape[0]):
+            y_batch = np.squeeze(self.y[b], axis=self.channel_axis - 1)
+            y_batch = np.expand_dims(y_batch, axis=self.channel_axis - 1)
+
+            self.y[b] = y_batch
+
+            if len(np.unique(self.y[b])) - 1 < self.min_objects:
+                invalid_batches.append(b)
+
+        invalid_batches = np.array(invalid_batches, dtype='int')
+
+        if invalid_batches.size > 0:
+            logging.warning('Removing %s of %s images with fewer than %s '
+                            'objects.', invalid_batches.size, self.x.shape[0],
+                            self.min_objects)
+
+        self.x = np.delete(self.x, invalid_batches, axis=0)
+        self.y = np.delete(self.y, invalid_batches, axis=0)
+        self.y_semantic_list = [np.delete(y, invalid_batches, axis=0)
+                                for y in self.y_semantic_list]
+
+        super(SemanticIterator, self).__init__(
+            self.x.shape[0], batch_size, shuffle, seed)
```

Comment:
```suggestion
        super(CropperIterator, self).__init__(
            self.x.shape[0], batch_size, shuffle, seed)
```

call `super` on the current class to run the parent class' `init` method. I think you can probably remove most of the above code and call the `SemanticIterator.__init__` right here instead.